cpp/inc/bond/core/maybe.h

```diff
@@ -45,17 +53,23 @@ class maybe
     }
 
     template <typename Allocator>
-    explicit maybe(Allocator& alloc)
+    explicit maybe(const Allocator& alloc)
         : _value(alloc),
           _nothing(true)
     {}
 
     template <typename Compare, typename Allocator>
-    explicit maybe(const Compare& comp, Allocator& alloc)
+    maybe(const Compare& comp, const Allocator& alloc)
```

Comment:
Consider checking where this is called to see whether we need such an overload at all.

Reply:
Looks like we do not need this anymore since we moved to `C++11`. Will send a separate PR after I confirm it works on all supported compilers.